internal/api/impl/proxy/proxy.go

```diff
@@ -132,18 +132,27 @@ func New(cfg config.DatasourceConfig, dashboardDAO dashboard.DAO, secretDAO secr
 func (e *endpoint) CollectRoutes(g *route.Group) {
 	if !e.cfg.Global.Disable {
 		g.ANY(fmt.Sprintf("/%s/:%s/*", utils.PathGlobalDatasource, utils.ParamName), e.proxySavedGlobalDatasource, false)
+		// allow direct datasource queries without extra path (e.g., from ClickHouse)
+		g.ANY(fmt.Sprintf("/%s/:%s", utils.PathGlobalDatasource, utils.ParamName), e.proxySavedGlobalDatasource, false)
+
 		// add route for SQLProxy kind to be able to POST directly to the datasource endpoint
 		g.POST(fmt.Sprintf("/%s/:%s", utils.PathGlobalDatasource, utils.ParamName), e.proxySavedGlobalDatasource, false)
```

Comment:
in that case, you can remove this route as the one you are adding is covering this one already